hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_order_data_manager.py

```diff
@@ -0,0 +1,284 @@
+        order_price: float,
+        client_order_i_d: int,
+        timestamp: float,
+    ):
+        self.order_status = order_status
+        self.quantity_released = quantity_released
+        self.quantity_remaining = quantity_remaining
+        self.side = side
+        self.fill_price = fill_price
+        self.order_price = order_price
+        self.client_order_i_d = client_order_i_d
+        self.timestamp = timestamp
+
+    def __eq__(self, other: "OrderStatusInfo"):
+        return (
+            self.order_status == other.order_status
+            and self.quantity_released == other.quantity_released
+            and self.quantity_remaining == other.quantity_remaining
+            and self.side == other.side
+            and self.fill_price == other.fill_price
+            and self.order_price == other.order_price
+            and self.client_order_i_d == other.client_order_i_d
+            and self.timestamp == other.timestamp
+        )
+
+
+class BloxrouteOpenbookOrderDataManager:
+    def __init__(self, provider: Provider, trading_pairs: List[str], owner_address: str, start_timeout: int = 10):
+        self._provider = provider
+        self._owner_address = owner_address
+        self._start_timeout = start_timeout
+
+        for index, trading_pair in enumerate(trading_pairs):
+            trading_pairs[index] = normalize_trading_pair(trading_pair)
+        self._trading_pairs = trading_pairs
+
+        self._order_books: Dict[str, OrderbookInfo] = {}
+        self._markets_to_order_statuses: Dict[str, Dict[int, List[OrderStatusInfo]]] = {}
+
+        self._started = False
+        self._ready = asyncio.Event()
+        self._is_ready = False
+
+        self._orderbook_polling_task: Optional[Task] = None
+        self._order_status_running_tasks: List[Task] = []
+        self._order_status_polling_task: Optional[Task] = None
+
+    async def ready(self):
+        await self._ready.wait()
+
+    @property
+    def is_ready(self):
+        return self._is_ready
+
+    @property
+    def started(self):
+        return self._started
+
+    async def start(self):
+        if not self._started:
+            self._started = True
+            await self._initialize_order_books()
+            self._orderbook_polling_task = asyncio.create_task(self._poll_order_book_updates())
+
+            await self._initialize_order_status_streams()
+
+            await asyncio.sleep(self._start_timeout)
+
+            self._is_ready = True
+            self._ready.set()
+
+    async def stop(self):
+        await self._provider.close()
+        if self._orderbook_polling_task is not None:
+            self._orderbook_polling_task.cancel()
+            self._orderbook_polling_task = None
+        for task in self._order_status_running_tasks:
+            if task is not None:
+                task.cancel()
+        self._order_status_running_tasks.clear()
+
+    async def _initialize_order_books(self):
+        await self._provider.connect()
```

Comment:
Best to do this in the `start()` block instead of here for consistency.

Reply:
For some reason, I need to do it in each function I use the provider in, or else the functions don't work 🤷